File: cpp/delete-characters-to-make-fancy-string.cpp
#include "string"

using namespace std;

class Solution {
public:
  string makeFancyString(string s) {
    string res = "";
    int count = 1;
    res += s[0];
    for (int i = 1; i < s.size(); i++) {
      if (s[i] == s[i - 1]) {
        count++;
      } else {
        count = 1;
      }
      if (count < 3) {
        res += s[i];
      }
    }
    return res;
  }
};

int main() {
  Solution s;

  return 0;
}

File: java/delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder sb = new StringBuilder();
        int count = 1;
        char prev = s.charAt(0);
        sb.append(prev);
        for (int i = 1; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == prev) {
                count++;
            } else {
                count = 1;
                prev = c;
            }
            if (count < 3) {
                sb.append(c);
            }
        }
        return sb.toString();      
      }
}
